aiidalab_widgets_base/nodes.py

```diff
@@ -207,16 +220,29 @@ def _find_called(cls, root):
 
     @classmethod
     def _find_outputs(cls, root):
-        assert isinstance(root, AiidaOutputsTreeNode)
         process_node = load_node(root.parent_pk)
-        outputs = {k: process_node.outputs[k] for k in process_node.outputs}
-        for key in sorted(outputs.keys(), key=lambda k: outputs[k].pk):
-            output_node = outputs[key]
-            if output_node.pk not in root.nodes_registry:
-                root.nodes_registry[output_node.pk] = cls._to_tree_node(
-                    output_node, name=f"{key}<{output_node.pk}>"
+        if root.namespace:
+            outputs = process_node.outputs[root.namespace]
+        else:
+            outputs = process_node.outputs
+
+        output_nodes = {key: outputs[key] for key in outputs}
+
+        for key in sorted(
+            output_nodes.keys(), key=lambda k: getattr(outputs[k], "pk", 0)
+        ):
+            node = output_nodes[key]
+
+            if isinstance(node, AttributeDict):
+                yield AiidaOutputsTreeNode(
+                    name=key, parent_pk=root.parent_pk, namespace=key
                 )
```

Comment:
This looks mostly good. What is the reason for not performing this conversion as part `._to_tree_node()`?

Reply:
Because after change `AiidaOutputsTreeNode` it no longer require have TreeNode object with `pk` as input, this is the reason cause this issue at first place. However  `._to_tree_node()` still try to use `pk` to initialize the TreeNode.

Reply:
Sure, but that logic (type check or check for the `.pk` attribute) could have also been done inside of `._to_tree_node()`. Either way, I think this is fine and refactoring the sequence would not improve readability significantly or perhaps at all.

Reply:
I guess it is possible to add an extra if-else in `to_tree_node` for nodes that are node the process type. But I think the current code is clear.